falcon/asgi/request.py

```diff
@@ -712,16 +715,33 @@ async def get_media(self):
 
         See also :ref:`media` for more information regarding media handling.
 
+        Note:
+            When ``get_media`` is called on a request with an empty body,
+            Falcon will let the media handler try to deserialize the body
+            and will return the value returned by the handler or propagate
+            the exception raised by it. To instead return a different value
+            in case of an exception by the handler, specify the argument
+            ``when_empty_fallback``.
+
         Warning:
             This operation will consume the request stream the first time
             it's called and cache the results. Follow-up calls will just
             retrieve a cached version of the object.
 
+        Args:
+            when_empty_fallback: Fallback value to return when there is no body
+                in the request and the media handler raises an error
+                (like in the case of the default JSON media handler).
+                By default, Falcon uses the value returned by the media handler
+                or propagates the raised exception, if any.
+
         Returns:
             media (object): The deserialized media representation.
         """
 
-        if self._media is not None or self.stream.eof:
+        if self._media_error is not None:
+            raise self._media_error
```

Comment:
My reasoning for not checking it is that after the first call, this function replays the previous call.

But I guess it may be strange

Reply:
on second thought, I think I agree with you. It makes better sense to consider it also on following calls, otherwise it's not a very useful parameter